flink-yarn/src/main/java/org/apache/flink/yarn/YarnResourceManager.java

```diff
@@ -135,7 +135,7 @@ public YarnResourceManager(
 			JobLeaderIdService jobLeaderIdService,
 			ClusterInformation clusterInformation,
 			FatalErrorHandler fatalErrorHandler,
-			@Nullable String webInterfaceUrl,
+			YarnResourceManagerConfiguration yarnResourceManagerConfiguration,
```

Comment:
I think the `ApplicationConstants.Environment.PWD` should not belong to the `YarnResourceManagerConfiguration `. We may replace it with "System.getEnv()" here.

Reply:
It's probably not a big problem, but I would try to avoid directly accessing system env from a component/plugin like `YarnResourceManager` or `YarnResourceManagerDriver` if possible. Injecting all the dependencies through parameters should be more testing friendly.